### PR TITLE
Add significant variable usage corner cases

### PR DESCRIPTION
**Pull request by erikerikson**
## What did you implement:

Add tests to ongoingly verify the cases addressed by issues #3627, #3740, #3711, and #3882 
For augmenting #4144 

## Todos:

- [x] Write tests
- [ ] Write documentation
- [ ] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/classes/Variables.test.js [hunk 1]
```diff
@@ -141,6 +142,135 @@ describe('Variables', () => {
...
+          val1: 'my value',
+          depVal: 'my value',
+          val0: 'my value',
+          val2: 'my value',
+        };
+        return expect(serverless.variables.populateObject(service).then((result) => {
+          expect(result.custom).to.eql(expected);
+        })).to.be.fulfilled;
+      });
+//       // REGARDING THESE CASES, SEE:
```
Comment (HyperBrain) [inline lib/classes/Variables.test.js]: As discussed, the comments to disable these tests for now are ok here 😃
